cpp/src/strings/padding.cu

```diff
@@ -38,6 +37,18 @@ namespace cudf {
 namespace strings {
 namespace detail {
 namespace {
+
+__device__ size_type compute_padded_size(string_view d_str,
```

Comment:
Wait, this is repeated from `pad_impl.cuh`.

Reply:
I only see it is used once so maybe don't put it in that header file.

Reply:
Argh. I forgot to remove this.

Reply:
This will be called from the strings-udf code in a future PR that is why it is being moved to a header.

Reply:
Fixed now. Also added doxygen to the functions in the new header.